File: 프로그래머스/lv2/12924. 숫자의 표현/숫자의 표현.java
class Solution {
    public int solution(int n) {
        int answer = 1;
        for(int i = 1; i <=n; i++) {
            int tmp = i;
            for(int j = i+1;j<=n;j++) {
                tmp += j;
                if(tmp == n) {
                    answer++;
                    break;
                }else if(tmp > n) {
                    break;
                }
            }
        }
        return answer;
    }
}

File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.py
def solution(n):
    answer = 1
    left = 1
    right = 1
    sumValue = 0
    
    while left <= right:
        if sumValue > n:
            sumValue -= left
            left += 1
        elif sumValue < n:
            sumValue += right
            right += 1
            if right > n:
                break
        else:
            answer += 1
            sumValue -= left
            left += 1
            
            
    return answer